sda-commons-client-jersey/README.md

```diff
@@ -228,3 +228,47 @@ org.glassfish.jersey.client.JerseyClientBuilder
 ```
 
 This works if the library that requires the other implementation does not rely on the Java ServiceLoader.
+
+
+### Consume API clients without catching exceptions
+
+The [`Suppress*`](./src/main/java/org/sdase/commons/client/jersey/proxy/annotation) annotations can be used to suppress
+specific `ClientRequestException` thrown in any defined API method where the return type is not a `Response`.
+There are configurable suppressors for 
+[HTTP errors](./src/main/java/org/sdase/commons/client/jersey/proxy/annotation/SuppressHttpErrors.java),
+[connection timeouts](./src/main/java/org/sdase/commons/client/jersey/proxy/annotation/SuppressConnectTimeoutErrors.java),
+[read timeouts](./src/main/java/org/sdase/commons/client/jersey/proxy/annotation/SuppressReadTimeoutErrors.java)
+and all other [processing errors](./src/main/java/org/sdase/commons/client/jersey/proxy/annotation/SuppressProcessingErrors.java).
+
+Suppressed exceptions are logged at info level.
+
+The following client will convert a _404 Not Found_ error to `null` so that the consumer can handle 
+an entity that is not found by checking if the return value is `null`.
+Any other error will lead to a _500 Internal Server Error_ because the `ClientRequestException` is converted in the
+[`ClientRequestExceptionMapper`](./src/main/java/org/sdase/commons/client/jersey/error/ClientRequestExceptionMapper.java).
+
+```java
+@Path("/examples")
+public interface ExampleClient {
+
+  @GET
+  @Path("/{exampleId}")
+  @Produces(MediaType.APPLICATION_JSON)
+  @SuppressHttpErrors(404)
```

Comment:
Do I understand it correctly that this annotation suppresses the exception and returns `null` instead?

Reply:
If yes: How about a name like `@ReturnNullForHttpErrors(404)`. On the other side I wonder if there aren't any REST Client Libraries for Java ecosystem that can already do this, instead of writing our own here.

Reply:
Yes, in this case only 404 Not Found errors. The _value_ is an array so that multiple status codes can be defined to be suppressed.

Reply:
As far as I know, JEE REST Clients are the most common clients in the Java ecosystem that use interfaces as alternative to fluent or other request builders. The clients based on the JEE standard are available in most frameworks like Dropwizard, Micronaut, Quarkus and monolithic JEE applications. One drawback we already worked around is that JEE uses the same exception types for client requests and server responses. Therefore we already added a layer around it so that developers carefully choose how to pass through client errors to their consumers. This PR just adds another layer of convenience to the common JEE implementation. It is based on the things we often repeat in our daily business. I think it can be transferred to other JEE implementations easily when we go for another framework. In my opinion this is far from „writing our own REST Client Library“.

`@ReturnNullFor*` is quite long compared to `@Suppress*`. What do the other @SDA-SE/sda-commons-maintainer  think? Is it more helpful to describe the result in the annotation? I'm fine with both variants.

Reply:
I'd vote for `@Suppress...` because I find it more common and the name contains what it does: suppressing an exception in case of http errors.

Reply:
What I miss about suppress is explicitly stating what happens instead. I had to read the code of this PR to understand it. Could also mean that another exception is thrown instead.

Reply:
There may be some inaccuracies in the wording but I think _suppress_ clearly states that no exception is thrown. What do you think about `@Suppress*ErrorsToNull`?

Reply:
I updated the PR to the pattern `@Suppress*ErrorsToNull`.